src/cpp/hu.bme.mit.inf.modes3.ESP32-sensor/railsensor_3/ESP32-sensor/src/3.cpp

```diff
@@ -0,0 +1,92 @@
+#define SENSOR_DISTANCE 7.0
+#define CONF_SSID "MoDeS3"
+#define PASS "LaborImage"
+
+#include <MQTT_JSON.hpp>
+#include <InfraSensor.hpp>
+#include <SensorStatemachine.hpp>
+#include <RealTimeSelect.hpp>
+#include <RealTimeLength.h>
+#include <realtimespeed.hpp>
+#include <string>
+#include <stdlib.h>
+
+struct Serl {
+  Serl() {
+    Serial.begin(9600);
+  };
+};
+Serl s;
+
+SensorStateMachine stateMachine;
+RealTimeSpeed RTSpeed(stateMachine.sensor[0].GetDexP(),stateMachine.sensor[0].DATAstack,stateMachine.sensor[1].GetDexP(),stateMachine.sensor[1].DATAstack);
+RealTimeLength Length(stateMachine.sensor[0].GetDexP(),stateMachine.sensor[0].DATAstack,stateMachine.sensor[1].GetDexP(),stateMachine.sensor[1].DATAstack);
+RealTimeSelect TrainSelect;
+WiFiClient wifi;
+MQTT_JSON send(&wifi);
+
+void setup() {
+  Serial.begin(9600);                           // Serial init
+
+  for(int i=0;i<2;i++){                         // Sensors input init
+    pinMode(stateMachine.sensor[i].GetPin(), INPUT);
+    stateMachine.sensor[i].Reset();
+  }
+  
+  WiFi.begin(CONF_SSID, PASS);                  // WiFi connection init
+  while (WiFi.status() != WL_CONNECTED)
+  {
+    delay(500);
+  }
+
+  send.client.setServer(MQTT_IP, MQTT_PT);      // MQTT connection init
+  while (!send.client.connected())
+  {
+    send.client.connect(DEVICE_NAME, MQTT_US, MQTT_PW);
+  }
+  send.client.setCallback(callback);
+  send.client.subscribe(DATA_CH);
+  send.client.subscribe(EVENT_CH);
+
+  TrainSelect.AddTrain("UNKNOWN",25.0);         // TrainSelect init
+  TrainSelect.AddTrain("Taurus",21.5);
+  TrainSelect.AddTrain("SNCF",18.5);
+  TrainSelect.AddTrain("Vagon",12.25);
+  TrainSelect.AddTrain("UNKNOWN",8.0);
+
+  stateMachine.Init(13,27);                    // stateMachine init
+
+  while (1)
+  {
+    Serial.println("Kakadu");

```

Comment:
For f.... sake guys... This is an open-source project, although, I admin we have funny commit messages from time to time, be a little more serious about the code you put on the internet.